### DISCUSSION TITLE
Contents of child component disappears in test after re-render of parent component

### DISCUSSION THREAD

**Discussion by DanielGSoftware** [livewire/livewire#8755] [7. Bugs (archived)]
### Livewire version

v3.5.4

### Laravel version

v11.2.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

When writing a test for a component that contains another component, the contents of this "child" component disappear after making a call in a test to a method. 

A simplified example:
```php
Livewire::test(ParentComponent::class)
    // Assertion successful
    ->assertSee('Child contents')
    ->call('foo')
    // Failed asserting that ... contains 'Child contents'.  
    ->assertSee('Child contents');
```

To reproduce this easily on your own, see linked GH repo.
I'm not of much use here unfortunately.. I couldn't figure out why this happens. Is this somehow intentional?

### Repository or Wirebox URL

https://github.com/DanielGSoftware/livewire-child-contents-dissapear-in-test

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
When the parent re- renders, the child will re-render as well.

Not sure what you expect?

Wait for a second before asserting for child content, if that is the default content etc

**Reply by DanielGSoftware** [reply]
Hey Joe, nice to see you again!

The fact that the child re-render when the parent re-renders too is indeed expected behaviour! The issue I'm having is that the content's of the child component disappear in my test after I call a public method (which indeed triggers a re-render). I don't expect this to happen, and also it's not in line with what I see on the actual page.

> Wait for a second before asserting for child content, if that is the default content etc

AFAIK there's not a way to do this in a component test and I believe also not something that's needed, I think Livewire waits untill the re-render/emits etc. are done before continuing with assertions again, right?

**Comment by squaredpx** [comment]
I'm experiencing the same issue, re-render is the expected behaviour but the child component contents disappears completely. In my case the parent component is a table, with each row as a child component. When I call de sort method in the parent component the action is triggered and the parent component is re-rendered but each row content is replaced by an empty `<tr wire:id="xxxxxx"></tr>` causing all "see" assertions to fail because there is no content inside the child components.
